Beam search tests are flaky

Once in a while the tests introduced in #20 will fail:

 ```
       # Begin Testing

        for vocab_num in [4, 8]:

            for decoder_fn in [RNNDecoder,

                               functools.partial(RNNDecoder2, use_tuple=False),

                               functools.partial(RNNDecoder2, use_tuple=True)]:

                decoder = decoder_fn(vocab_num=vocab_num, hidden_size=HIDDEN_SIZE)

                decoder.hybridize()

                decoder.initialize()

                for beam_size, bos_id, eos_id, alpha, K in [(2, 1, 3, 0, 1.0),  (4, 2, 3, 1.0, 5.0)]:

                    scorer = BeamSearchScorer(alpha=alpha, K=K)

                    for max_length in [10, 20]:

                        sampler = BeamSearchSampler(beam_size=beam_size, decoder=decoder, eos_id=eos_id,

                                                    scorer=scorer, max_length=max_length)

                        for batch_size in [1, 2, 5]:

                            states = decoder.begin_state(batch_size)

                            inputs = mx.nd.full(shape=(batch_size,), val=bos_id)

                            samples, scores, valid_length = sampler(inputs, states)

                            samples = samples.asnumpy()

                            scores = scores.asnumpy()

                            valid_length = valid_length.asnumpy()

                            for i in range(batch_size):

                                max_beam_valid_length = int(np.round(valid_length[i].max()))

                                step_states = _fetch_step_states(states, i, beam_size)

                                step_input = bos_id

                                npy_samples = _npy_beam_search(decoder, scorer, step_input, step_states,

                                                               eos_id, beam_size, max_length)

>                               assert_allclose(npy_samples, samples[i, :, :max_beam_valid_length])

E                               AssertionError: 

E                               Not equal to tolerance rtol=1e-07, atol=0

E                               

E                               (mismatch 20.8333333333%)

E                                x: array([[1., 2., 0., 2., 0., 2., 0., 2., 2., 0., 2., 3.],

E                                      [1., 2., 0., 2., 0., 2., 0., 2., 2., 2., 2., 3.]])

E                                y: array([[1, 2, 0, 2, 0, 2, 2, 0, 2, 0, 2, 3],

E                                      [1, 2, 0, 2, 0, 2, 2, 0, 2, 2, 0, 3]], dtype=int32)


tests/unittest/test_beam_search.py:184: AssertionError

```


Should we fix the seed for these tests? If so we can simply add the `@pytest.mark.seed(1)` decorator.

--- a/tests/unittest/test_beam_search.py
+++ b/tests/unittest/test_beam_search.py
@@ -1,10 +1,13 @@
-import numpy as np
-import mxnet as mx
-from mxnet.gluon.rnn import RNNCell
-from mxnet.gluon import nn, HybridBlock
-from gluonnlp.model import BeamSearchSampler, BeamSearchScorer
-from numpy.testing import assert_allclose
 import functools
+
+import mxnet as mx
+import numpy as np
+import pytest
+from mxnet.gluon import HybridBlock, nn
+from mxnet.gluon.rnn import RNNCell
+from numpy.testing import assert_allclose
+
+from gluonnlp.model import BeamSearchSampler, BeamSearchScorer
 
 
 def test_beam_search_score():
@@ -23,6 +26,7 @@ def test_beam_search_score():
             assert_allclose(scores.asnumpy(), sum_log_probs.asnumpy() / lp, 1E-5, 1E-5)
 
 
+@pytest.mark.seed(1)
 def test_beam_search():
     def _get_new_states(states, new_states, sel_beam_ids, use_new_states):
         if isinstance(states, list):

